trie/committer.go

```diff
@@ -104,87 +97,93 @@ func (c *committer) commit(n node, db *Database, force bool) (node, error) {
 	case *shortNode:
 		// Commit child
 		collapsed := cn.copy()
-		if _, ok := cn.Val.(valueNode); !ok {
-			childV, err := c.commit(cn.Val, db, false)
+
+		// If the child is fullnode, recursively commit.
+		// Otherwise it can only be hashNode or valueNode.
+		if _, ok := cn.Val.(*fullNode); ok {
```

Comment:
Can't the child be another shortnode? 
Let's say you have a small trie. The root will be a shortnode, when you start inserting, you will have only shortnodes until they start getting crowded, no?

Reply:
The child of a shortnode can't be the shortnode. If so, they can be compressed into a single one. Look at here https://github.com/ethereum/go-ethereum/blob/master/trie/trie.go#L299

> The root will be a shortnode, when you start inserting, you will have only shortnodes until they start getting crowded

When you insert some entries, the shortnode will be converted into a fullnode, the inserted entry will be created as the shortnode.

Reply:
Hi,
Somehow I ended up on this git email distribution list.  Can somebody please remove me??  I get emails three times a day with this subject:  

[ethereum/go-ethereum] trie: polish committer (#21351)

Reply:
Sorry, removed.